core/trino-main/src/main/java/io/trino/operator/output/RleAwarePositionsAppender.java

```diff
@@ -52,7 +52,7 @@ public RleAwarePositionsAppender(BlockPositionIsDistinctFrom isDistinctFromOpera
     @Override
     public void append(IntArrayList positions, Block source)
     {
-        // RleAwarePositionsAppender should be used with FlatteningPositionsAppender that makes sure
+        // RleAwarePositionsAppender should be used with UnnestingPositionsAppender that makes sure
         // append is called only with flat block
         checkArgument(!(source instanceof RunLengthEncodedBlock));
```

Comment:
Let's have an error message with some details about the block here (separate commit)